### PR TITLE
add new Game method to load game from JSON or DB

### PR DESCRIPTION
**Pull request by michaeljb**
cc @jenf

### CODE REVIEW

File: assets/app/view/game_page.rb [hunk 1]
```diff
@@ -53,20 +53,7 @@ def load_game
       return if game_id == @game&.id &&
         ((!cursor && @game.actions.size == @num_actions) || (cursor == @game.actions.size))
 
-      # Some Hotseat games don't have player ids, use names instead.
-      players = @game_data['players'].map { |p| [p['id'] || p['name'], p['name']] }.to_h
-      # Back compatibility, make hotseat games continue to work after the change to play names
-      if actions&.first&.dig(:entity_type) == 'player' && actions.first[:entity].is_a?(String)
-        players = @game_data['players'].map { |p| [p['name'], p['name']] }.to_h
```
Comment (michaeljb) [inline assets/app/view/game_page.rb:60]: Is it important to keep this part for backward compatibility?

Comment (tobymao) [reply assets/app/view/game_page.rb:60]: na

File: lib/engine/game/base.rb [hunk 2]
```diff
@@ -30,6 +30,46 @@
 
 module Engine
   module Game
+    def self.load(source, at_action: nil, **kwargs)
+      hash_game = nil
+      db_game = nil
+      case source
```
Comment (tobymao) [inline lib/engine/game/base.rb:36]: maybe game_data = case source
when Hash, String

{ }...
else
{] ...
end

Comment (michaeljb) [reply lib/engine/game/base.rb]: Done in 9d89a026f6e0afd0eed322b20ea3cdf3dcfc5531

**Review by tobymao — APPROVED** [review]

**Pull request merged**